CMakeLists.txt

```diff
@@ -214,17 +239,33 @@ if(BUILD_DX11 AND WIN32)
         file(TO_CMAKE_PATH ${COMPILED_SHADERS} COMPILED_SHADERS)
     endif()
 
-    list(APPEND LIBRARY_SOURCES ${COMPILED_SHADERS}/BC6HEncode_EncodeBlockCS.inc)
-
     if(NOT USE_PREBUILT_SHADERS)
-        add_custom_command(
-            OUTPUT "${COMPILED_SHADERS}/BC6HEncode_EncodeBlockCS.inc"
-            MAIN_DEPENDENCY "${PROJECT_SOURCE_DIR}/DirectXTex/Shaders/CompileShaders.cmd"
-            DEPENDS ${SHADER_SOURCES}
-            COMMENT "Generating HLSL shaders..."
-            COMMAND ${CMAKE_COMMAND} -E env CompileShadersOutput="${COMPILED_SHADERS}" CompileShaders.cmd > "${COMPILED_SHADERS}/compileshaders.log"
-            WORKING_DIRECTORY "${PROJECT_SOURCE_DIR}/DirectXTex/Shaders"
-            USES_TERMINAL)
+	    if(MINGW)
+		    message(STATUS "Generating HLSL shaders...")
```

Comment:
Why not just update the CompilerShaders.cmd script directly?

Where is your 'fxc' coming from in this case?

Reply:
Note the script should default to just 'fxc.exe' if it can't find the Windows SDK locations through environment variables already

Reply:
> Why not just update the CompilerShaders.cmd script directly?

I did install the latest Windows SDK but that does not have a command-line, and it can be difficult to access this from the MSYS2 environment particularly as the paths contain spaces.  I did base the CMake macros partly on the .cmd script although.  I also have to be honest in saying that I probably am more familiar with CMake then the .cmd command-lines shell.  Besides, I'm not sure I like having to invoke CMake and a cmd shell.  It's not as flexible as I required.

> Where is your 'fxc' coming from in this case?

The fxc program I obtained is downloadable from the MSYS2  ( https://www.msys2.org/ ) distribution in a pacman-package and the home page for the program is https://github.com/mozilla/fxc2 .

Reply:
Perhaps, it may help if I explain what I'm trying to do.  I'm writing a PKGBUILD script that builds DirectXTex into a redistributable package that I contribute to the MSYS2 ( http://www.msys2.org ) project.  I can't count on the computer building this redistributable package to have the Windows SDK installed along with the global path.  MSYS2 tries to be a self-contained Unix-like environment for building Windows software.  I was able to use the changes to CMakeLists.txt to build DirectXTex with Directx11 run-time support and even went so far as to build the ddsviewer.

Reply:
Are you saying the wine FXC is always present in MSYS2?

Reply:
There is a fxc.exe  but that is not always installed.  It can be installed simply by using "pacman" to install the appropriate fxc2 packages (this is strictly OPTIONAL).  What I will probably do is add "fxc2" as a buildtime dependency meaning it would be required to build DirectXTex package with DirectX 11 support.

Reply:
I have gone ahead and modified the hlsl_compile so that it will try to use Microsoft's fxc syntax and if that fails, then try the subset for fxc2.